scipy/stats/_continuous_distns.py

```diff
@@ -6123,6 +6125,43 @@ def _stats(self, b, moments='mv'):
     def _entropy(self, c):
         return 1 + 1.0/c - np.log(c)
 
+    def fit(self, data, *args, **kwds):
+        data = np.asarray(data)
+
+        floc = kwds.pop('floc', None)
+
+        # borrowed from distn_infrastructure
+        # puts shape into f0
```

Comment:
Yes, the default fit method supports `f0`, `fb`, and `fix_b`, and then that loop standardizes it by putting the shape in kwds as `f0`. I suppose it may be simpler just to check if those three names are in the kwds key dictionary, and then pull whichever one that is out.

Reply:
What a nightmare. But yes, try to replace all this with one line.